core-rust/state-manager/src/staging/cache.rs

```diff
@@ -89,27 +113,22 @@ impl ExecutionCache {
         }
     }
 
-    pub fn execute<S, T>(
+    pub fn execute<S: RootStore, T: FnOnce(&StagedStore<S>) -> TransactionReceipt>(
         &mut self,
         root_store: &S,
         parent_hash: &AccumulatorHash,
         new_hash: &AccumulatorHash,
         transaction: T,
```

Comment:
Just a thought - but could we call this eg execute_transaction or something? As it's a lazy method now? :)